$OpenBSD: patch-ext_json_ext_fbuffer_fbuffer_h,v 1.1 2015/01/08 18:54:20 jeremy Exp $

In ruby 2.2, rb_str_new is a macro instead of a function, so manually
expand the FBUFFER_PAIR macro so rb_str_new gets two arguments.

--- ext/json/ext/fbuffer/fbuffer.h.orig	Fri Nov 28 18:48:49 2014
+++ ext/json/ext/fbuffer/fbuffer.h	Fri Nov 28 18:49:06 2014
@@ -172,7 +172,7 @@ static FBuffer *fbuffer_dup(FBuffer *fb)
 
 static VALUE fbuffer_to_s(FBuffer *fb)
 {
-    VALUE result = rb_str_new(FBUFFER_PAIR(fb));
+    VALUE result = rb_str_new(FBUFFER_PTR(fb), FBUFFER_LEN(fb));
     fbuffer_free(fb);
     FORCE_UTF8(result);
     return result;
